PsevdoTwwiterApp/src/main/webapp/resources/js/script.js

```diff
@@ -0,0 +1,638 @@
+                    }
+                    break;
+                case 'createdAt':
+                    if (!post.createdAt || Object.prototype.toString.call(post.createdAt) !== '[object Date]') {
+                        return false;
+                    }
+                    break;
+                case 'photoLink':
+                    if (post.photoLink && typeof post.photoLink !== 'string') {
+                        return false;
+                    }
+                    break;
+                case 'hashTags':
+                    if (post.hashTags) {
+                        if (!post.hashTags.every(tag => typeof tag === 'string')) {
+                            return false;
+                        }
+                    }
+                    break;
+                case 'likes':
+                    if (post.likes.length !== 0) {
+                        if (!post.likes.every(like => typeof like === 'string')) {
+                            return false;
+                        }
+                    }
+                    break;
+                default:
+                    return false;
+            }
+            return true;
+        });
+    }
+
+    add(post = {}) {
+        post.likes = [];
+        if (Model.validate(post)) {
+
+            this.data.push(post);
+            this.data.sort(Model.comparator_BY_DATE);
+            let o = JSON.stringify(post);
+            localStorage.setItem(post.id, o);
+            return true;
+        }
+        return false;
+    }
+
+    edit(id, tweet) {
+
+        if (!Model.validate(tweet, Object.keys(tweet))) {
+            return false;
+        }
+
+        let editingTweet = this.get(id);
+
+        for (let param in tweet) {
+            editingTweet[param] = tweet[param];
+        }
+        localStorage.removeItem(id);
+        localStorage.setItem(id, JSON.stringify(editingTweet));
+        return true;
+    }
+
+    remove(id) {
+        if (typeof id === 'string') {
+            let index = this.data.findIndex(post => post.id === id);
+
+            if (index !== -1) {
+                this.data.splice(index, 1);
+                localStorage.removeItem(id);
+                return true;
+            }
+        }
+        return false;
+    }
+
+
+    like(id = '', user = '') {
+        let post = this.get(id);
+        let index = post.likes.findIndex(like => like === user);
+        if (index === -1) {
+            post.likes.push(user);
+            localStorage.removeItem(id);
+            localStorage.setItem(id, JSON.stringify(post));
+            return true;
+        }
+
+        return false;
+    }
+    dislike(id = '', user = '') {
+        let post = this.get(id);
+        let index = post.likes.findIndex(like => like === user);
+        if (index !== -1) {
+            post.likes.splice(index, 1);
+            localStorage.removeItem(id);
+            localStorage.setItem(id, JSON.stringify(post));
+            return true;
+        }
```

Comment:
You can simplify  `like` and `dislike` to one function with a boolean argument. It will reduce code.